packages/server/lib/browsers/chrome.ts

```diff
@@ -532,20 +541,24 @@ export = {
       debug('closing remote interface client')
 
       // Do nothing on failure here since we're shutting down anyway
-      launchedBrowser.browserCriClient.close().catch()
+      browserCriClient.close().catch()
+      browserCriClient = undefined
 
       debug('closing chrome')
 
       originalBrowserKill.apply(launchedBrowser, args)
     }
 
-    const criClient = await browserCriClient.attachToTargetUrl('about:blank')
+    const pageCriClient = await browserCriClient.attachToTargetUrl('about:blank')
 
-    await this._setAutomation(browserCriClient, criClient, automation)
+    await this._setAutomation(pageCriClient, automation, browserCriClient.closeCurrentTarget)
+
+    await Promise.all([
+      this._maybeRecordVideo(pageCriClient, options, browser.majorVersion),
+      this._handleDownloads(pageCriClient, options.downloadsFolder, automation),
+    ])
```

Comment:
just curious, why move the navigate & download order?

Reply:
Both of these can be done in parallel and aren't dependent upon the navigation to occur. It's a slight performance boost.